src/index.ts

```diff
@@ -133,6 +79,28 @@ swdcTracker.trackEditorAction = async (params: EditorActionParams): Promise<any>
   return await swdcTracker.spTracker.trackUnstructEvent(_editorActionPayload, contexts);
 }
 
+/**
+ * @param jwt - the authorization token
+ * @param params - the UI Interaction params
+ */
+swdcTracker.trackUIInteraction = async (params: UIInteractionParams): Promise<any> => {
+
+  // build the contexts and event payload
+  const _uiInteractionPayload: any = await new UIInteraction(params).buildPayload();
+  const contexts: any = await buildContexts(params);
+
+
```

Comment:
I think we could package lines 93-103 into a function that gets called by each `track<Event_name?`  function.

Something like:
```
function sendEvent(event_payload: any, contexts: any) {
  if (isTestMode()) {
     // test mode - console log the event
     return testEvent(_uiInteractionPayload, contexts);
   }
  return await swdcTracker.spTracker.trackUnstructEvent(event_payload, contexts);
}
```

That should DRY up the implementation to look like:
```
sendEvent(
  await new UIInteraction(params).buildPayload(), 
  await buildContexts(params)
);
```

Reply:
yeah, definitely. i'll reduce it down

Reply:
it's updated